android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt

```diff
@@ -82,20 +105,47 @@ class MainViewModel : IpnViewModel() {
     }
   }
 
+  fun maybeShowVpnPermissionLauncher() {
```

Comment:
Perhaps rename to "showVPNPermissionLauncherIfUnauthorized"? Or whatever makes it clearer what it does. :)